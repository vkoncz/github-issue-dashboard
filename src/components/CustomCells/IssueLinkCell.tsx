import type { ICellRendererParams } from 'ag-grid-community';

type IssueLinkCellParams = Omit<ICellRendererParams, 'value'> & { value: string };

export function IssueLinkCell({ value }: IssueLinkCellParams) {
  return (
    <span>
      <a href={value} target="_blank" rel="noreferrer">
        Open issue
      </a>
    </span>
  );
}
