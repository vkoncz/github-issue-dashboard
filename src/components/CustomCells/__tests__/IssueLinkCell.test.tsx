import { render, screen } from '@testing-library/react';

import { IssueLinkCell } from '../IssueLinkCell';

describe('IssueLinkCell', () => {
  it('should contains a link', () => {
    const { container } = render(
      // @ts-ignore
      <IssueLinkCell value="https://" />,
    );

    const link = screen.getByRole('link');

    expect(link).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should handle undefine value without generate link', async () => {
    // @ts-ignore
    const { container } = render(<IssueLinkCell />);

    const link = screen.queryByRole('link');

    expect(link).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
