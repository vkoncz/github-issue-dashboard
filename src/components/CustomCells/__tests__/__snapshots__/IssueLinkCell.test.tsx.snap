// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueLinkCell should contains a link 1`] = `
<div>
  <span>
    <a
      href="https://"
      rel="noreferrer"
      target="_blank"
    >
      Open issue
    </a>
  </span>
</div>
`;

exports[`IssueLinkCell should handle undefine value without generate link 1`] = `<div />`;
